src/Components/WithCurrentTime.tsx

```diff
@@ -0,0 +1,50 @@
+import React from "react"
```

Comment:
I really like this component, basically any other component that depends on current time (and wants to accordingly re-render as time passes), will have a similar set up with `setInterval`, etc.

So I think this is an elegant solution (I'm still quite new to these sorts of patterns), so thank you!